python/pipeline/meso.py

```diff
@@ -519,55 +565,301 @@ def make(self, key):
+                    f,
+                    scan[:, :, :, :, start_idx:end_idx],
+                    field_id=0,
+                    y=slice(skip_rows, -skip_rows),
+                    x=slice(skip_cols, -skip_cols),
+                    channel=0,
+                    kwargs=kwargs,
+                )
+                if type(scan) == np.ndarray:
+                    y_shifts_fragment = np.zeros(end_idx - start_idx)
+                    x_shifts_fragment = np.zeros(end_idx - start_idx)
+                for frames, chunk_y_shifts, chunk_x_shifts in results:
+                    y_shifts_fragment[frames] = chunk_y_shifts
+                    x_shifts_fragment[frames] = chunk_x_shifts
+
+                x_shift_fragments[n, start_idx:end_idx] = x_shifts_fragment
+                y_shift_fragments[n, start_idx:end_idx] = y_shifts_fragment
+
+                # Use the beginning of the fragment and weight it from 0 to 1 to be summed with
+                # overlap area added to end_idx. This will be used to take the weighted mean.
+                if start_idx != 0:
+                    start_taper_weights = np.linspace(0, 1, overlap_size)
+                    x_shift_fragments[n, start_idx : start_idx + overlap_size] = (
+                        x_shift_fragments[n, start_idx : start_idx + overlap_size]
+                        * start_taper_weights
+                    )
+                    y_shift_fragments[n, start_idx : start_idx + overlap_size] = (
+                        y_shift_fragments[n, start_idx : start_idx + overlap_size]
+                        * start_taper_weights
+                    )
+                if end_idx_base != scan.shape[-1]:
+                    end_taper_weights = np.linspace(1, 0, overlap_size)
+                    x_shift_fragments[n, end_idx_base:end_idx] = (
+                        x_shift_fragments[n, end_idx_base:end_idx] * end_taper_weights
+                    )
+                    y_shift_fragments[n, end_idx_base:end_idx] = (
+                        y_shift_fragments[n, end_idx_base:end_idx] * end_taper_weights
+                    )
+
+            second_x_shifts, second_y_shifts, _ = galvo_corrections.process_xy_motion(
+                np.nansum(x_shift_fragments, axis=0),
+                np.nansum(y_shift_fragments, axis=0),
+                key,
+            )
+
+            x_shifts = second_x_shifts + old_x_shifts
+            y_shifts = second_y_shifts + old_y_shifts
 
         # Create results tuple
         tuple_ = key.copy()
+        tuple_.pop("motion_correction_method")
+        tuple_.pop("channel")
         tuple_["motion_template"] = template
         tuple_["y_shifts"] = y_shifts
         tuple_["x_shifts"] = x_shifts
         tuple_["outlier_frames"] = outliers
         tuple_["y_std"] = np.std(y_shifts)
         tuple_["x_std"] = np.std(x_shifts)
 
+        # Create record of motion correction method
+        method_used_key = key.copy()
+        method_used_key.pop("channel")
```

Comment:
why is the channel not in the MotionMethodUsed?  won't that cause confusion if correction is performed on more than one channel?  Or are all these tables designed to bottleneck to a single channel at this point?